Mhagdorn reread the paper (#319)
the-teachingRSE-project/competencies: competencies.md

One of the crucial skills of RSEs is their mental proximity to research.
They embrace being part of a larger community which,
despite friendly competition, shares the common goal of gaining knowledge

> to disseminate it.

Reviewer: In my opinion, here you change the meaning. is this intended?

Author: I did change the meaning but I think we don't know what the motivation of people is. So I thought a more neutral formulation would be good

Reviewer 2: The meaning is a bit changed (from "gaining knowledge just to gain it" to "gaining knowledge to disseminate it"), but I think this is clearer and more accurate.

Author: I think this meaning of the statement is not clear from the sentence. We'd need to reformulate that then. I will have a think about it when I have a chance to. Maybe more towards the end of the week. Feel free to come up with a different formulation.

Reviewer: ```suggestion
of pursuing and disseminating knwledge.
```

| \gls{TEACH}      | Should be able to perform simple peer-to-peer on-boarding tasks.                                                              | Should be able to explain logical components and the general architecture to other RSEs.                                              | Should be able to effectively communicate about all high-level parts of the project.                                                                                                |
| \gls{PM}         | Should be aware of the employed \gls{PM} method.                                                                    | Should be able to use and adapt the employed \gls{PM} method.                                                                               | Should be able to design and adapt the employed \gls{PM} method.                                                                                                                          |

> | \gls{USERS}      | Should be able to communicate with SEs and researchers of the project.                                                               | Same as junior RSE, and be able to interpret the feedback.                                                                            | Same as senior, and should also be able to effectively take feedback into account when steering the project.                                                                        |

Reviewer 3: ```suggestion
| \gls{USERS}      | Should be able to communicate with both users and SEs on the project, on topics of the research and SE.                                                               | Same as junior RSE, and be able to interpret the feedback.                                                                            | Same as senior, and should also be able to effectively take feedback into account when steering the project.                                                                        |
```

    \\\hline
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    \gls{TEAM}&

>     Must be able to work effectively with researchers they are potentially developing code for.&

Reviewer: Maybe qualify it a bit?  E.g. If not developing code for themselves, then they must....

Author: good idea, done

RSEs interested in developing their (\gls{TEACH}) skill can focus on teaching the next generation of researchers and/or RSEs and will play
a vital role in improving the quality of research software.
They need to have a good understanding of all RSE competencies relevant to their domain and

> additionally should have teaching experience and training in didactic and pedagogy.

Reviewer 2: Even though the Greek word would indeed be "didactic" (διδακτική), in English the term is "didactic**s**", right? "didactic" should be an adjective only.

```suggestion
additionally should have teaching experience and training in didactics and pedagogy.
```

Author: I think the spellchecker complained and a quick google did not come up with didactics. it certainly sounds right to me.

At the intersection of data science [@SSIDataScience] and data-focused RSEs,
the complex way of solving problems utilising \ac{ML} calls for this separate specialisation.

> #### Legacy RSEs

Reviewer: Any motivation for this change? Not that I'm against it....

Reviewer 2: Maybe even put it at the end, as in "we don't really want to look here"?

Author: I thought the infa, hpc and ml RSE should stay together since they are somewhat related

Research software may have evolved over generations of researchers without change management or governance processes, while software "ecosystems" (e.g., programming languages, frameworks, operating systems) constantly evolve.
This may lead to the emergence of legacy code that is still actively used.
To safeguard continued usability and adoption,
these RSEs have experience in working with legacy code,
and are competent in the application of software stacks that are no longer part of the general curricula (e.g., \gls{COBOL} or \gls{Fortran}).
Adaption of existing, large-scale codebases to evolving dependencies (\gls{DIST}) or changing hardware (\ac{HPC}; see the HPC-RSE point below)
may require mastery in refactoring techniques and in the usage of specialised code transformation tools.

#### Web-development RSE
This RSE is skilled in web applications, front- and/or backend, and/or building
and using APIs, for example for research data portals or big research projects.
Ideally, this RSE should also have knowledge about (web) accessibility to allow a broad
range of researchers or even the public to use the resulting applications.

> Therefore, a deep knowledge of web development skills is a required additional skill for this RSE.

Reviewer: Here, the qualification that *Web* is an *Additional* Skill is not explicitely stated anymore.

Author: good point, put it back

 that RSEs that choose explicitly a science-supporting role outside of research will not be eligible for funding
 under the statutes of many funding organisations that require at least a PhD.

To alleviate this and to give RSEs in leadership positions a means to become eligible for funding themselves,
 since completion of scientific training is often a requirement [@DFG_50_01],
 we see two possible parts of a solution.
One is to allow for doctorates primarily based on software contributions to the scientific community.
Secondly, we propose the introduction of new, standardised certificates like those in @subsec:examplecareer,
 and to officially accept them as PhD-equivalent concerning eligibility to be a \ac{PI}.
Beyond this discussion, a diverse set of publications on the topic RSE teaching is already in the making.

Within this set, we will work next on how to institutionalise education.
In that publication, we will detail how we organise our institutions
 and what qualifications our teachers need to have in order to effectively communicate our values.
We will put forward ideas on how to build up bachelor's and master's programmes,

>  of which a glimpse can already be found in the appendix.

Reviewer: While at it, maybe a forward reference.

Author: good plan

that provide this education will be the topic of a follow-up paper.

# Contribution details {-}

> Heidi Seibold came up with the original idea for the deRSE23 workshop.

Reviewer: A well, it was not only the idea, she also organised it, and her initial structure guided the questions we asked...

Author: That's stated in the next sentence

Reviewer 2: I was wondering if I should at some point ask this, but since a lot has happened since this declaration was first written (and having the [Ship of Theseus](https://en.wikipedia.org/wiki/Ship_of_Theseus) paradox in mind): where do we now draw the line regarding the original draft preparation?

I am fine with anything, but I thought I would shortly spotlight that some of us (hoping it is not just me) are in a gray zone. I definitely don't want to trigger any issues here.

Author: I think Heidi does deserve the credit. I just wanted to tidy up this section a little by removing repetitions. I also think the word _conceive_ was somewhat overused. Also it would need to be _conceive of_ otherwise it implies pregnancy.

Reviewer 2: To avoid confusion: I am not talking about the original idea etc, I am talking about the rest of the statement (since the paragraph was touched here).

Reviewer 2: Based on the discussion on today's meeting, let's address the complete CRediT statement in a separate PR.
